appservice/src/deploy/deploy.ts

```diff
@@ -80,9 +81,17 @@ export async function deploy(client: SiteClient, fsPath: string, context: IDeplo
                 if (!(await fse.pathExists(fsPath))) {
                     throw new Error(localize('pathNotExist', 'Failed to deploy path that does not exist: {0}', fsPath));
                 }
-
-                if (config.linuxFxVersion && /^(tomcat|wildfly)/i.test(config.linuxFxVersion)) {
+                const javaRutime = client.isLinux ? config.linuxFxVersion : config.javaContainer;
+                if (javaRutime && /^(tomcat|wildfly|jboss)/i.test(javaRutime)) {
                     await deployWar(context, client, fsPath);
+                } else if (javaRutime && /^java/i.test(javaRutime)) {
+                    // For Java SE runtime, need rename the artifact to app.jar
+                    let javaArtifact: string = fsPath;
+                    if (path.basename(javaArtifact) !== "app.jar") {
```

Comment:
`.jar` files can be quite large.  This copy could potentially take a long time and we would also be writing it in a temporary folder where it would be taking up space.  Unless explicitly deleted, I don't believe temp files are deleted automatically either. (source https://www.makeuseof.com/tag/why-temp-files-not-deleted-automatically/)

I think that we could just change the name of the metadata of the zip file as it's being zipped [here](https://github.com/microsoft/vscode-azuretools/blob/main/appservice/src/deploy/runWithZipStream.ts#L55)

Reply:
@nturinski Thanks for your advices, I'll refactor the logic to do the rename in package.

Reply:
I think I have a potential solution for this refactor.

We could let `runWithZipStream` have an optional parameter that is of type `{ [key: string]: string }`.  The key would be the real file name and the value would be the desired file name.

So for example, if the object is `{ main.jar: app.jar }`, this [line](https://github.com/microsoft/vscode-azuretools/blob/main/appservice/src/deploy/runWithZipStream.ts#L55) could look like this:
```
const zipFileMetadata= { main.jar : "app.jar" } // this would be passed in as a parameter
zipFile.addFile(path.join(fsPath, file), zipFileMetadata[file] || file);
```

Reply:
@nturinski Thanks a lot for your advices, I create a new `deployJar` method because renaming the artifact to app.jar is only the requirement for java se deployment, and there seems no similar requirement for other languages cause only java requires an artifact file while others require to deploy the project, so I'm not sure whether I should add extra parameters to the shared `runWithZipStream`.